### ISSUE TITLE
Static lease does not work

### ISSUE THREAD

**Issue by p53** [lxc/lxd#5106]
# Required information

 * Distribution: bionic
 * Distribution version: 18.04.1
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.15.0-34-generic
   * LXC version:
   * LXD version: 3.0.1-0ubuntu1~18.04.1
   * Storage backend in use: directory

# Issue description

container doesn't assign static lease, altough it is visible in container configuration
to understand what i want to achive: i want to be able to assign static ip to eth0 (I don't want to add additional interfaces because then i would need to configure /etc/network/interfaces inside them to come up and i would like to work it out of box without additional configuration), another issue i found is that when i don't remove eth0 from default config i cannot assign static ip to eth0 (which is in default profile) because it says no such device

# Steps to reproduce

 1. lxc profile device remove default eth0
 2. lxc network create test ipv4.address=10.0.3.1/24 ipv4.nat=true
 3. lxc launch ubuntu:xenial test1
 4. lxc config device add test1 eth0 nic parent=test nictype=bridged ipv4.address=10.0.3.23
 5. lxc restart test1

```
lxc network list-leases test
+----------+-------------------+------------+---------+
| HOSTNAME |    MAC ADDRESS    | IP ADDRESS |  TYPE   |
+----------+-------------------+------------+---------+
| fafa     | 00:16:3e:8b:98:ce | 10.0.3.24  | STATIC  |
+----------+-------------------+------------+---------+
| test     | 00:16:3e:dd:0c:e5 | 10.0.3.23  | STATIC  |
+----------+-------------------+------------+---------+
| test1     | 00:16:3e:dd:0c:e5 | 10.0.3.239 | DYNAMIC |
+----------+-------------------+------------+---------+

lxc list
+------+---------+-------------------+----------------------------------------------+------------+-----------+
| NAME |  STATE  |       IPV4        |                     IPV6                     |    TYPE    | SNAPSHOTS |
+------+---------+-------------------+----------------------------------------------+------------+-----------+
| fafa | RUNNING |                   |                                              | PERSISTENT | 0         |
+------+---------+-------------------+----------------------------------------------+------------+-----------+
| test1 | RUNNING | 10.0.3.239 (eth0) | fd42:ae81:35a6:dac0:216:3eff:fedd:ce5 (eth0) | PERSISTENT | 0         |
+------+---------+-------------------+----------------------------------------------+------------+-----------+

cat /var/lib/lxd/networks/test/dnsmasq.leases 
1538691823 00:16:3e:dd:0c:e5 10.0.3.239 test1 *
duid 00:01:00:01:23:46:96:3d:72:af:a5:b7:42:5c

ls -al /var/lib/lxd/networks/test/dnsmasq.hosts/
total 8
drwxr-xr-x 2 root root 4096 Oct  4 20:32 .
drwx--x--x 3 root root 4096 Oct  4 21:16 ..

lxc config show test1 --expanded
architecture: x86_64
config:
  image.architecture: amd64
  image.build: "20181004_07:42"
  image.description: Ubuntu xenial amd64 (20181004_07:42)
  image.distribution: ubuntu
  image.name: ubuntu-xenial-amd64-default-20181004_07:42
  image.os: ubuntu
  image.release: xenial
  image.serial: "20181004_07:42"
  image.variant: default
  volatile.base_image: 950e32336b6551a7cc619bbc318d97bf64ebc26be4e8daa971539a1b0e616f0d
  volatile.eth0.hwaddr: 00:16:3e:dd:0c:e5
  volatile.eth0.name: eth0
  volatile.idmap.base: "0"
  volatile.idmap.next: '[{"Isuid":true,"Isgid":false,"Hostid":165536,"Nsid":0,"Maprange":65536},{"Isuid":false,"Isgid":true,"Hostid":165536,"Nsid":0,"Maprange":65536}]'
  volatile.last_state.idmap: '[{"Isuid":true,"Isgid":false,"Hostid":165536,"Nsid":0,"Maprange":65536},{"Isuid":false,"Isgid":true,"Hostid":165536,"Nsid":0,"Maprange":65536}]'
  volatile.last_state.power: RUNNING
devices:
  eth0:
    ipv4.address: 10.0.3.23
    nictype: bridged
    parent: test
    type: nic
  root:
    path: /
    pool: default
    type: disk
ephemeral: false
profiles:
- default
stateful: false
description: ""

lxc info test1 --show-log
Name: test1
Remote: unix://         
Architecture: x86_64
Created: 2018/10/04 21:20 UTC
Status: Running      
Type: persistent
Profiles: default
Pid: 18355
Ips:
  eth0: inet    10.0.3.239      vethQLS0SQ                
  eth0: inet6   fd42:ae81:35a6:dac0:216:3eff:fedd:ce5   vethQLS0SQ
  eth0: inet6   fe80::216:3eff:fedd:ce5 vethQLS0SQ
  lo:   inet    127.0.0.1  
  lo:   inet6   ::1            
Resources:                                               
  Processes: 9              
  CPU usage:  
    CPU usage (in seconds): 0                                                                          [27/1873]
  Memory usage:                               
    Memory (current): 25.24MB                 
    Memory (peak): 30.39MB                                                 
  Network usage:                                                 
    eth0:                                                                     
      Bytes received: 4.60kB
      Bytes sent: 2.06kB                  
      Packets received: 40                 
      Packets sent: 16                             
    lo:   
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0    
      Packets sent: 0
                
Log:  

```
# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

**Comment by stgraber** [comment]
I believe this has already been fixed in master and the fix should also be in 3.0.2.

I tested on 3.5 and it's working as expected here, I get a lease in the dnsmasq lease file and the container gets it over DHCP.

**Comment by p53** [comment]
yup tested on 3.0.2, it works, thanks!
